File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict
def solution(participant, completion):
    answer = ''
    player=defaultdict(int)
    for p in participant:
        player[p]+=1
    for c in completion:
        player[c]-=1
    
    sorted_player = sorted(player.items(),key=lambda x:-x[1])
    
    answer = sorted_player[0][0]
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
#include <unordered_map>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    unordered_map<string,int> player;
    
    for(string p : participant){
        player[p]++;
    }
    for(string c : completion){
        player[c]--;
    }
    
    for(auto x:player){
        if (x.second==1) {
            answer=x.first;
            break;
        }
    }
    
    
    
    return answer;
}